### ISSUE TITLE
Field of duration type changes magnitude after round-trip through Parquet +shaping

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#2428]
Repro is with `zq` commit 7182f9e.

```
$ echo '{duration:4.266ms}' | zq -f parquet - > duration.parquet

$ zq -z -i parquet duration.parquet 
{duration:4266000}

$ zq -i parquet duration.parquet | zq -z 'type d={duration:duration}; put .=shape(d)' -
{duration:49d9h} (=d)
```

@nwt identified the responsible piece of code as [https://github.com/brimsec/zq/blob/7182f9e87105ec8429350ba040bff2c23ccb2f7e/expr/cast.go#L132.](https://github.com/brimsec/zq/blob/7182f9e87105ec8429350ba040bff2c23ccb2f7e/expr/cast.go#L132.)

The current proposal from @mccanne around which we've established consensus:

> I suggest we use parquet INTERVAL, truncate to millisecond precision, and assume a month is 30 days of ns.  This way we are consistent in and out Z for any duration up to milllisecond precision.
> 
> If someone wants to do it differently then can use Z arithmetic and cast to int64.

**Comment by philrz** [comment]
There's a workaround for this, which is that a user could write some Zed to cast these to an integer before writing the data out as Parquet, then reverse the cast when reading it back in. Since that gives us at least some guidance to offer when a user bumps into this, we may keep this one in the Icebox until we have enough uptake in Parquet use that someone hits it.

**Comment by philrz** [comment]
Note to self: I'd like to do a comprehensive boomerang test where I take a large ZSON sample data set, output it as Parquet, then re-read it back with Zed shaping, then compare the original ZSON with the re-shaped ZSON and look for remaining diffs. Since we know this duration problem is going to cause a giant diff, I'm holding off on doing that test until this one is fixed.

**Comment by mccanne** [comment]
Elevating priority...

**Comment by philrz** [comment]
I confirmed via binary search that this was fixed starting with Zed commit 0b6233b, which is associated with #3448. Updating the original repro steps to current Zed syntax, now we see that the `duration` value completes the round-trip through Parquet and shaping.

```
$ zq -version
Version: v0.33.0-81-g0b6233ba

$ echo '{duration:4.266ms}' | zq -f parquet - > duration.parquet

$ zq -z -i parquet duration.parquet
{duration:4266000}

$ zq -i parquet duration.parquet | zq -z 'type d={duration:duration} yield shape(d)' -
{duration:4.266ms}(=d)
```

Thanks @nwt!
